python/hsml/core/dataset_api.py

```diff
@@ -53,11 +53,9 @@ def upload(
 
         ```python
 
-        import hopsworks
+        from hsml.core import dataset_api
 
-        project = hopsworks.login()
-
-        dataset_api = project.get_dataset_api()
+        dataset_api = dataset_api.DatasetApi()
```

Comment:
This won't work. Users should login before using dataset_api.
In order to get rid of the `hopsworks` reference, we can replace
`project = hopsworks.login()`
with
`conn = hsml.connection(project="my-project")`
`dataset_api = conn.get_dataset_api()`